suzieq/engines/pandas/network.py

```diff
@@ -200,23 +205,37 @@ def _find_addr_arp(self, addr: str, **kwargs) -> pd.DataFrame:
         ifdf = self._get_table_sqobj('interfaces', start_time='') \
             .get(namespace=namespaces, hostname=hostnames,
                  ifname=ifaces)
+        # We don't explicitly look for only local-only entries because
+        # that precludes us addresses that were not remote in the past.
         macdf = self._get_table_sqobj('macs', start_time='') \
                     .get(namespace=namespaces, hostname=hostnames,
-                         macaddr=macaddrs, columns=['default'],
-                         local=True)
+                         macaddr=macaddrs,
+                         columns=['namespace', 'hostname', 'vlan', 'macaddr',
+                                  'oif', 'remoteVtepIp', 'flags', 'active',
+                                  'timestamp'])
 
         for arp_row in arpdf.itertuples():
-            row_end_time = arp_row.timestamp + timedelta(seconds=60)
+            row_end_time = arp_row.next_ts or self.iobj.end_time
             tmpres = {}
             if getattr(arp_row, 'error', None):
                 continue
             if arp_row.oif.endswith('-v0'):
-                # Handle VRR interfaces in Cumulus
-                oif = arp_row.oif[:-3]
-            else:
-                oif = arp_row.oif
+                # Ignore VRR interfaces in Cumulus, we got them via the
+                # normal interface entry (without -v0)
+                continue
+            oif = arp_row.oif
 
             active = getattr(arp_row, 'active', True)
+            if (arp_row.remote and
+                    arp_row.hostname not in active_local_hosts):
+                continue
+
+            # If an entry goes from local to being deleted, remove it
+            if not active and arp_row in active_local_hosts:
+                active_local_hosts.remove(arp_row.hostname)
+
+            if active and not arp_row.remote:
+                active_local_hosts.add(arp_row.hostname)
 
             # We ignore the start time because the interface may not have
```

Comment:
This comment seems a leftover. Probably it should be moved up to line 205

Reply:
Ack